.gitignore

```diff
@@ -5,3 +5,4 @@
 .pub/
 
 build/
+pubspec.lock
```

Comment:
usually lock files are not ignored, is there an idea behind why it should be ignored?

Reply:
Lock files should be ignored in library packages, they bring no value. After all, it is up to end users of this library to resolve dependencies, ie. according to the constraints in pubspec. Lock files are only recommended for end users (applications). See https://dart.dev/guides/libraries/private-files#pubspeclock for more info.